runtime/drivers/postgres/postgres.go

```diff
@@ -76,8 +82,25 @@ func (d driver) Open(instanceID string, config map[string]any, st *storage.Clien
 	if instanceID == "" {
 		return nil, errors.New("postgres driver can't be shared")
 	}
-	// actual db connection is opened during query
+
+	conf := &ConfigProperties{}
+	if err := mapstructure.WeakDecode(config, conf); err != nil {
+		return nil, fmt.Errorf("failed to decode config: %w", err)
+	}
+
+	dsn := conf.ResolveDSN()
+	if dsn == "" {
+		return nil, fmt.Errorf("database_url or dsn not provided")
+	}
+
+	// Open DB handle
+	db, err := sqlx.Open("pgx", dsn)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open connection: %w", err)
+	}
```

Comment:
Keeping a permanently open handle only for pings seems excessive. Postgres struggles with many concurrent open connections, so we should be economical with it. Better to just open the handle as part of ping until we have other use cases that require a persistently open connection.